rskj-core/src/main/java/org/ethereum/vm/program/Program.java

```diff
@@ -453,6 +481,10 @@ public void createContract2(DataWord value, DataWord memStart, DataWord memSize,
     }
 
     private void createContract( RskAddress senderAddress, byte[] nonce, DataWord value, DataWord memStart, DataWord memSize, RskAddress contractAddress, boolean isCreate2) {
+        // [1] FETCH THE CODE FROM THE MEMORY
+        byte[] programCode = memoryChunk(memStart.intValue(), memSize.intValue());
```

Comment:
if possible I'd try to make this check after the `isNotCovers()` one, where we were initially reading memory chunk (this reading op. can be quite heavy)

Reply:
so there's a reason.. ok then